### PR TITLE
Require RSpec >= 3.12 and drop Pundit < 2.0 support

### LINKED ISSUE #67: Drop support for older versions of Pundit  and RSpec
* [ ] Remove `permit_action(:action_name, *arguments)` and `forbid_action(:action_name, *arguments)`
* [ ] Remove "Testing Actions With Arguments" section from the Readme.
* [ ] Remove deprecation warnings from Readme.
* [ ] Update `pundit-matchers.gemspec` ( `s.add_runtime_dependency` and `s.add_development_dependency`)
* [ ] Release new major version of Pundit Matchers

### PR DESCRIPTION
**Pull request by tagliala**
Also:
- Bump major version
- Replace RSpec-rails with RSpec
- Do not require pundit (it was not used)
- Add debug and documentation files to gitignore
- Update Gemfile.lock

Close #67

### CODE REVIEW

**Comment by tagliala** [comment]
Hi @chrisalley , dropping support for legacy libraries could be a good thing so we can drop compatibility for policy with arguments and that would simplify the migration from `forbid_action` to `forbid_actions` with a single parameter

File: pundit-matchers.gemspec [hunk 1]
```diff
@@ -12,8 +12,7 @@ Gem::Specification.new do |s|
   s.require_paths = ['lib']
   s.homepage    = 'https://github.com/punditcommunity/pundit-matchers'
   s.license     = 'MIT'
-  s.add_runtime_dependency 'rspec-rails', '>= 3.0.0'
```
Comment (tagliala) [inline pundit-matchers.gemspec:15]: I do not understand why `rspec-rails` was a requirement, if this is only using `rspec`

File: pundit-matchers.gemspec [hunk 2]
```diff
@@ -12,8 +12,7 @@ Gem::Specification.new do |s|
...
-  s.add_development_dependency 'pundit', '~> 1.1', '>= 1.1.0'
```
Comment (tagliala) [inline pundit-matchers.gemspec:16]: also, this is ignored when using the gem in a third-party and pundit is not used to test matchers. I've dropped it

**Comment by chrisalley** [comment]
If the gem doesn't depend on `pundit` or `rspec-rails`, it makes sense to remove these from the Gemspec. From memory (it was a long time ago now), I added these more as a documentation of what the gem was aiming to support. It should be sufficient to simply list the target gem versions in the Readme.

Instead of this paragraph: 

> As of Pundit Matchers 2, Ruby 3 is a requirement. Pundit Matchers also requires that both the [rspec-rails](https://rubygems.org/gems/rspec-rails) (v3.0+) and [pundit](https://rubygems.org/gems/pundit) (v1.1+) gems are installed. We will drop support for older versions of these gems in Pundit Matchers 3.

We could instead simply list the target versions, e.g.

> Pundit Matchers expects that the application you're testing is using a software stack consisting of:
>
> * Ruby 3+
> * Pundit 2+
> * RSpec 3+

Or some similar text to this.

If we don't have any dependencies on Rails, it makes sense to target `rspec` rather than `rspec-rails` (even if Pundit itself seems to expect that you're using Rails).

**Comment by chrisalley** [comment]
The change looks good.

* This commit has been merged into the `main` branch.
* I've created a release (`v2.2.0`) with your other recent changes.
* I've set up a seperate branch for the 2.x release line (`v2.x`).
* 3.x code can be merged into `main`. We can do the `v3.0.0` release once the other issues related to the milestone are met.

**Pull request merged**